com.unity.netcode.gameobjects/Editor/Simulator/NetworkTypeView.cs

```diff
@@ -168,5 +172,11 @@ void UpdateLiveIfPlaying()
                 m_NetworkSimulator.UpdateLiveParameters();
             }
         }
+        
+        void SetSimulatorConfiguration(NetworkSimulatorConfiguration configuration)
+        {
+            m_NetworkSimulator.SimulatorConfiguration = configuration;
+            EditorUtility.SetDirty(m_NetworkSimulator);
```

Comment:
Should we use `[Undo](https://docs.unity3d.com/ScriptReference/Undo.html).RecordObject` instead? To allow for undo?

Reply:
That sounds like a good approach, let me check if I can add it.

Reply:
It turns out that using `RecordObject` didn't work right out of the box, I've tried it in many different ways, but because of the way we were updating the value it didn't work.

So I've found that the right way of updating values like this, is through SerializedObjects / SerializedProperties ([source](https://docs.unity3d.com/ScriptReference/PrefabUtility.RecordPrefabInstancePropertyModifications.html)):

> _Note that the recommended method of enabling instances of Prefabs to record changes is to use SerializedProperty and SerializedObject. This enables instances to record changes and automatically includes changes in the undo system_

So using a SerializedProperty everything worked just fine, I just had to listen for the Undo event to update the UI accordingly.

You can check the specific [commit here](https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/pull/2045/commits/e221d23ae3df8b821050d3507f1ae6f69e309e11).